src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,26 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  setSelectTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setSelectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(todo?.userId).then(setUser);
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    getUser(todo.userId).then(setUser);
  }, [todo.userId]);
```
- React Hook useEffect has a missing dependency: 'todo?.userId'.
- And todo is required so you don't need to use optional chaining